connection.go

```diff
@@ -170,11 +170,13 @@ func (sc *snowflakeConn) Close() (err error) {
 	sc.stopHeartBeat()
 
 	// ensure transaction is rollbacked
-	_, err = sc.exec(context.Background(), "ROLLBACK", false, false, nil)
+	ctx := context.TODO()
```

Comment:
seems like maybe these methods should be taking contexts? but maybe a battle for another day

Reply:
These are the `database/sql` methods... which for some reason don't take a context. Seems like the fix here is to release local resources and then kick off a goroutine to close down the session on the snowflake side.